src/Role/Controller/UpdateRoleController.php

```diff
@@ -0,0 +1,84 @@
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\Role\Controller;
+
+use OpenApi\Attributes\JsonContent;
+use OpenApi\Attributes\Put;
+use OpenApi\Attributes\RequestBody;
+use Pimcore\Bundle\StudioBackendBundle\Controller\AbstractApiController;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\DatabaseException;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\NotFoundException;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Parameters\Path\IdParameter;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Response\DefaultResponses;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Response\SuccessResponse;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Config\Tags;
+use Pimcore\Bundle\StudioBackendBundle\Role\MappedParameter\UpdateRoleParameter;
+use Pimcore\Bundle\StudioBackendBundle\Role\Schema\DetailedRole;
+use Pimcore\Bundle\StudioBackendBundle\Role\Schema\UpdateRole;
+use Pimcore\Bundle\StudioBackendBundle\Role\Service\RoleServiceInterface;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constants\HttpResponseCodes;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constants\UserPermissions;
+use Pimcore\Bundle\StudioBackendBundle\Util\Traits\PaginatedResponseTrait;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @internal
+ */
+final class UpdateRoleController extends AbstractApiController
+{
+    use PaginatedResponseTrait;
+
+    public function __construct(
+        SerializerInterface $serializer,
+        private readonly RoleServiceInterface $roleUpdateService
+    ) {
+        parent::__construct($serializer);
+    }
+
+    /**
+     * @throws NotFoundException|DatabaseException
+     */
+    #[Route('/role/{id}', name: 'pimcore_studio_api_role_update', methods: ['PUT'])]
+    #[IsGranted(UserPermissions::USER_MANAGEMENT->value)]
+    #[Put(
+        path: self::API_PATH . '/role/{id}',
+        operationId: 'updateRoleById',
+        summary: 'Update role by id.',
+        tags: [Tags::Role->value]
+    )]
+    #[IdParameter(type: 'Role')]
+    #[SuccessResponse(
+        description: 'Updated data.',
+        content: new JsonContent(ref: DetailedRole::class)
+    )]
+    #[RequestBody(
+        content: new JsonContent(ref: UpdateRole::class)
+    )]
+    #[DefaultResponses([
+        HttpResponseCodes::NOT_FOUND,
+    ])]
+    public function updateRoleById(int $id, #[MapRequestPayload] UpdateRoleParameter $roleUpdate): JsonResponse
+    {
+        return $this->jsonResponse(
+            $this->roleUpdateService->updateRoleById($id, $roleUpdate)
```

Comment:
For e.g. update the asset, we call the updateService, but it actually does not return anything.

The thing is, that we should not return the updated data immediately, because there could be save listeners to the object. It would be better to call the "get" again instead of returning the updated data directly. 
At least that was our way of thinking for the assets.
Should be consistent with the rest. 

This would also reduce code duplication in case of the additional attributes since it only then needs to be in the "get" operation

Reply:
Sounds reasonable! I will change it, and also for the Update User Endpoint.

Reply:
Fixed https://github.com/pimcore/studio-backend-bundle/pull/233/commits/591b2061b8c1d9734af536c53f27b62ce4a8b590